srcs/app/config/analyser/Analyser.cpp

```diff
@@ -1,18 +1,20 @@
 #include <parser/Analyser.hpp>
 
-Analyser::Analyser(void) {}
+Analyser::Analyser(void) : line_(1) {}
 
 Analyser::~Analyser(void) {}
 
-Analyser::SyntaxError::SyntaxError(const std::string &error, size_t line)
-	: line_(line) {
-	std::ostringstream str;
-	str << error << " in line " << line_;
-	lineerr_ = str.str();
+char	Analyser::SyntaxError::lineerror_[200] = {};
+
+Analyser::SyntaxError::SyntaxError(const std::string &error, size_t line) {
+	// if this didn't need to not throw, we would use std strings or string
+	// streams
+	snprintf(lineerror_, sizeof(lineerror_), "%s in line %zu",
+	error.c_str(), line);
```

Comment:
Great idea to use snprintf. Thanks for the comments explaining why it is necessary.

Reply:
I made it by hand [here](https://github.com/42AGV/webserv/pull/118/commits/22d421eac53e3574bf419ebf9a7caebfb31d7a04), but then I told myself that it was ridiculous to do such kind of artisan micromanagement.